Dpp 1305 6 1309 integration

tom-gough opened the pull request.

Commit:
Updates following integration

Commit:
Merge branch 'TransfersBaseBranch_2' into DPP-1305-6-1309-integration

Commit:
Merge branch 'TransfersBaseBranch_2' into DPP-1305-6-1309-integration

Commit:
Fixed column names / definitions on transfers transient view ddl

Commit:
Fixed spelling mistakes, added tasklog definition, removed unneeded columns

Commit:
Integration work - fixes to view ddl, transfer date, copy mappings, acceptance test copy mapping call

Commit:
Merge branch 'TransfersBaseBranch_2' into DPP-1305-6-1309-integration

Commit:
Working Scenario for Transfers 1254

Commit:
transfer levy debit rule moved to it's own class

Commit:
Removed unneeded test data model

Commit:
Fixed transfer assertions and data collector to assert against specific employer accounts

scottywakefield commented on src/AcceptanceTesting/SpecByExample/SFA.DAS.Payments.AcceptanceTests/Assertions/TransfersAssertions.cs:
```diff
@@ -16,8 +15,16 @@ public static void ValidateTransfers(TransfersBreakdown breakdown, List<Transfer
+                            x => PeriodNameHelper.GetStringDateFromLongPeriod(x.CollectionPeriodName) == periodValue.PeriodName
+                            && x.SendingAccountId == breakdown.SendingEmployerAccountId
+                            && x.ReceivingAccountId == line.ReceivingEmployerAccountId
+                    ).Amount : 0;
```
would it be better to capture the output of `transferResults.FirstOrDefault` as a variable to then perform the null check on and resultant assignment to `currentResultAmount`?

tom-gough replied:
yes it would, I'll change it

Commit:
Updated transfer assertions to avoid duplicate linq logic

Commit:
Renamed RuleResult to ActualRuleResult

carlbm committed:
Removed PK from transient transferpayments.payments
Removed unneeded Funding account it from payments
Removed id from AccountTransfers transient table

carlbm committed:
Added unit tests to ensure that the transfer amount and payment amount are identical

carlbm committed:
Fixed issue with indexes on transient
Fixed mismatch drop/create on transient tables
Removed fundingaccountid property
